lib/SIL/IR/SIL.cpp

```diff
@@ -398,7 +398,8 @@ bool AbstractStorageDecl::exportsPropertyDescriptor() const {
     llvm_unreachable("should be definition linkage?");
   }
 
-  if (isUnsupportedKeyPathValueType(getValueInterfaceType())) {
+  auto *env = getDeclContext()->getGenericEnvironmentOfContext();
+  if (isUnsupportedKeyPathValueType(getValueInterfaceType(), env)) {
```

Comment:
It would be better to do the mapTypeIntoContext() here instead of passing a GenericEnvironment down to the function

Reply:
I tried doing that a while ago, but that seemingly caused other things to break. I didn't really investigate as this seemed to work.